src/Altinn.AccessManagement/Controllers/ResourceOwnerAPI/AppsInstanceDelegationController.cs

```diff
@@ -0,0 +1,234 @@
+    /// <param name="appInstanceDelegationRequestDto">The request model</param>
+    /// <param name="resourceId">The resource id</param>
+    /// <param name="instanceId">The instance id</param>
+    /// <returns>Result</returns>
+    [HttpPost]
+    [Route("v1/apps/instancedelegation/{resourceId}/{instanceId}")]
+    [Authorize(Policy = AuthzConstants.PLATFORM_ACCESS_AUTHORIZATION)]
+    [Consumes(MediaTypeNames.Application.Json)]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(AppsInstanceDelegationResponseDto), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(void), StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(typeof(void), StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult> Delegation([FromBody] AppsInstanceDelegationRequestDto appInstanceDelegationRequestDto, [FromRoute] string resourceId, [FromRoute] string instanceId)
+    {
+        AppsInstanceDelegationRequest request = _mapper.Map<AppsInstanceDelegationRequest>(appInstanceDelegationRequestDto);
+        
+        request.ResourceId = resourceId;
+        request.InstanceId = instanceId;
+
+        List<AttributeMatch> performedBy = GetOrgAppFromToken(Request.Headers["PlatformAccessToken"]);
+                
+        request.PerformedBy = performedBy;
+
+        Result<AppsInstanceDelegationResponse> serviceResult = await _appInstanceDelegationService.Delegate(request);
+
+        return serviceResult.IsProblem ? serviceResult.Problem?.ToActionResult() : Ok(_mapper.Map<AppsInstanceDelegationResponseDto>(serviceResult.Value));
+    }
+
+    /// <summary>
+    /// delegating app from the platform token
+    /// </summary>
+    private static List<AttributeMatch> GetOrgAppFromToken(string token)
+    {
+        List<AttributeMatch> performedBy = new List<AttributeMatch>();
+
+        if (!string.IsNullOrEmpty(token))
+        {
+            var handler = new JwtSecurityTokenHandler();
+            var jwtSecurityToken = handler.ReadJwtToken(token);
+            var appidentifier = jwtSecurityToken.Claims.FirstOrDefault(c => c.Type == AltinnXacmlConstants.MatchAttributeIdentifiers.AppAttribute);
+            performedBy.Add(new AttributeMatch { Id = AltinnXacmlConstants.MatchAttributeIdentifiers.OrgAttribute, Value = jwtSecurityToken.Issuer });
+            if (appidentifier != null)
+            {
+                performedBy.Add(new AttributeMatch { Id = appidentifier.Type, Value = appidentifier.Value });
+            }
+        }
+
+        return performedBy;
+    }
+
+    /// <summary>
+    /// Gets app instance delegation
+    /// </summary>
+    /// <param name="appInstanceDelegationRequestDto">The request model</param>
+    /// <returns>Result</returns>
+    [HttpPost]
+    [Authorize(Policy = AuthzConstants.POLICY_APPS_INSTANCEDELEGATION)]
+    [Route("v1/apps/instancedelegation/query")]
```

Comment:
Trenger opprydding for v1 og v2 alternativene, til v1 men med {resourceId}/{resourceInstanceId} path'er,  og fjerne modeller som ikke lenger er i bruk.

Reply:
Fjernet ekstra endepunkt.